### DISCUSSION TITLE
serialize和deserialize的翻譯

### DISCUSSION THREAD

**Discussion by husohome** [python/python-docs-zh-tw#547] [Add Term]
### 初步確認

- [X] 我已經確認過 [術語列表](https://github.com/python/python-docs-zh-tw/wiki/%E8%A1%93%E8%AA%9E%E5%88%97%E8%A1%A8) 中沒有相關術語。
- [X] 我已經確認過 [Python 官方術語列表](https://docs.python.org/zh-tw/3/glossary.html) 中沒有相關術語。

### 術語原文

serialize

### 術語翻譯

串聯化

### 說明

serialize 比較常聽到「序列化」但是教育部好像很明確說是「串聯化」
不是很確定哪個比較容易懂
老實說我覺得兩個詞都很奇怪
一直很想翻譯成「轉成文字儲存」

反之 deserialize 比較常聽到「反序列化」，但是「反」完全不符合台灣中文使用的邏輯
教育部樂詞網將 deserialization 翻譯成「還原序列化」，我覺得相當合理
但是如果將 deserialization 翻成 「還原序列化」不就表示 serialize 翻成「序列化」 就可以了嗎?

想知道各位大神看法

### 參考資料

https://terms.naer.edu.tw/detail/d95a51d50b23a7814ac55b86c6caf7fa/?startswith=zh&seq=1
https://terms.naer.edu.tw/detail/5f202bb0037cfc09fa338aac902e17ba/?startswith=zh&seq=1

**Comment by mattwang44** [comment]
> 「反」完全不符合台灣中文使用的邏輯

想了解更多支持這個說法的閱讀資料 🙏🏽

**Reply by husohome** [reply]
其實不瞞各位大神說這是小弟自己的語感

教育部樂詞網裡
#### 「反」在台灣的翻譯裡面，會對應到字根主要是 `anti-` (e.g. anti-gay 反同), `trans-` (e.g., transfat 反式脂肪), 以及 `contra`
https://terms.naer.edu.tw/search/?csrfmiddlewaretoken=sMcuj4j0RBW6o6c0Hef7KqxxpGr2jHG5kHekh0j5LvW1lW82Kc9cLjotqwt3nAI9&match_type=phrase&query_op=&query_field=title&query_term=%E5%8F%8D&filter_term=&filter_field=&filter_op=&filter_bool=&filter_term=&filter_field=&filter_op=&filter_bool=&filter_term=&filter_field=&filter_op=&filter_bool=&filter_term=&filter_field=&filter_op=&filter_bool=

所以我覺得「反」在比較偏學術的詞彙中，比較會是「反對」、「排序相反」、「於對立面」的意思
deserialize中的 `de-` 的意思是其實是 `undo` serialization 或者甚至是 `remove` serialization 的意思，並不是「反對」，也不是把 serialize 的結果的文字倒過來。

#### `de` 這個字根最常見的意思有好幾個
1. 去掉, e.g., detransition (把 transition的效果「去掉」) -- 這種時候會翻成「去」https://terms.naer.edu.tw/search/?csrfmiddlewaretoken=sMcuj4j0RBW6o6c0Hef7KqxxpGr2jHG5kHekh0j5LvW1lW82Kc9cLjotqwt3nAI9&match_type=phrase&query_op=&query_field=title&query_term=de&filter_term=&filter_field=&filter_op=&filter_bool=&filter_term=&filter_field=&filter_op=&filter_bool=&filter_term=&filter_field=&filter_op=&filter_bool=&filter_term=&filter_field=&filter_op=&filter_bool=
-- 我覺得把 deserialize 中的 `de-` 想成去掉序列化的效果，翻成「去序列化」也說得通
2. 使分離 e.g., detach
3. 以及在非常少見的情況下，`de` 的確可以翻成還原，請見「信號還原」，這個就非常接近 `decode` 或者 `undo` 的意思
https://terms.naer.edu.tw/search/?query_term=%E9%82%84%E5%8E%9F&query_field=title&query_op=&match_type=phrase&filter_bool=and&filter_term=%27%E9%9B%BB%E6%A9%9F%E5%B7%A5%E7%A8%8B%E5%90%8D%E8%A9%9E%27&filter_field=subcategory_1.raw&filter_op=term&tabaction=browse

#### 小結論?
所以小的認為 deserialize 翻成可能可以翻 「還原序列化」或「去序列化」

**Comment by xchux** [comment]
依照一些外部文件翻譯，也是使用「還原序列化」
- [microsoft Azure](https://learn.microsoft.com/zh-tw/python/api/azure-cognitiveservices-vision-customvision/azure.cognitiveservices.vision.customvision.prediction.models.customvisionerrorexception?view=azure-python)

但我個人過往是看到「反序列化」居多，所以也習慣了

**Comment by husohome** [comment]
> ### 術語原文
> serialize
> ### 術語翻譯
> 串聯化

我後來想了想，這裡的串聯好像是講電池的「串聯」(series connection)、「並聯」(parallel connection) 那個
所以這裡的串聯化 (serialization) 可能是指使電池變成 series 
所以目前比較支持把 serialize 翻成 「序列化」
